src/sessionTypes/cliAuth.spec.js

```diff
@@ -0,0 +1,394 @@
+        .returns(authenticationSession);
+
+      cliAuth = new CliAuth(sdk);
+    });
+
+    it('appends the supplied sdk to the class instance', function() {
+      expect(cliAuth._sdk).to.deep.equal(sdk);
+    });
+
+    it('calls the auth0 Authentication constructor with the proper arguments', function() {
+      expect(authentication).to.be.calledOnce;
+      expect(authentication).to.be.calledWith({
+        domain: 'ndustrial.auth0.com',
+        clientID: cliAuth._sdk.config.auth.clientId
+      });
+    });
+
+    it('sets an initial, empty session info state', function() {
+      expect(cliAuth._sessionInfo).to.be.an('object');
+      expect(cliAuth._sessionInfo).to.be.empty;
+    });
+  });
+
+  describe('logIn', function() {
+    let cliAuth;
+
+    context('a successful login', function() {
+      let expectedResponse;
+      let getApiToken;
+      let password;
+      let promise;
+      let username;
+
+      beforeEach(function() {
+        sdk = {
+          config: {
+            audiences: {
+              contxtAuth: fixture.build('audience'),
+              facilities: fixture.build('audience')
+            },
+            auth: {
+              clientId: faker.internet.password()
+            }
+          }
+        };
+        password = faker.internet.password();
+        username = faker.internet.email();
+        expectedResponse = {
+          accessToken: faker.internet.password(),
+          expiresIn: faker.random.number({ min: 100, max: 1000 }),
+          tokenType: 'Bearer'
+        };
+
+        authenticationSession = {
+          loginWithDefaultDirectory: this.sandbox
+            .stub()
+            .yields(null, expectedResponse)
+        };
+        this.sandbox
+          .stub(auth0, 'Authentication')
+          .returns(authenticationSession);
+
+        getApiToken = this.sandbox
+          .stub(CliAuth.prototype, '_getApiToken')
+          .resolves('Contxt Authentication successful.');
+
+        cliAuth = new CliAuth(sdk);
+
+        promise = cliAuth.logIn(username, password);
+      });
+
+      it("calls the 'loginWithDefaultDirectory' function with the username and password and correct audience", function() {
+        expect(
+          authenticationSession.loginWithDefaultDirectory
+        ).to.be.calledOnce;
+        expect(
+          authenticationSession.loginWithDefaultDirectory
+        ).to.be.calledWith({
+          username,
+          password,
+          audience: cliAuth._sdk.config.audiences.contxtAuth.clientId
+        });
+      });
+
+      it('updates the session info', function() {
+        return promise.then(() => {
+          expect(cliAuth._sessionInfo).to.deep.equal({
+            accessInfo: expectedResponse
+          });
+        });
+      });
+
+      it('returns a fulfilled promise with a success message', function() {
+        return promise.then((response) => {
+          expect(response).to.equal('Contxt Authentication successful.');
```

Comment:
If you don't change `_getApiToken` to return the token, you should make this `'Contxt Authentication successful.'` a generated variable so it is more clear exactly what is going on

Reply:
Updated to return the token 👍